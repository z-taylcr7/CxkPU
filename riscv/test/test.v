module adder(X, Y, Cin, Cout, S);

	input [0:1] X;
	input [0:1] Y;
	input Cin;
	output reg [0:1]S;
	output reg Cout;

	always @(X , Y , Cin) begin
   		 {Cout , S} = X + Y + Cin;
	end
endmodule

module testBench();
    reg [0:1] A , B;
    reg Cin;
    output [0:1]S;
    output Cout;
    
    adder U(
    .X(A),
    .Y(B),
    .Cin(Cin),
    .Cout(Cout),
    .S(S)
    );
    
    initial begin
        A = 0;
        B = 2;
        Cin = 1;
        #10 A = 1;
        #10 Cin = 1;
        #10 B = 0;
        #20 A = 3;
        #30 Cin = 0;
        #40 B = 3;
        #50 Cin = 1;
    end
    
    initial begin
        $monitor($time,,,"part:%b  %b", S, Cout);
        #70
        $finish;
    end
/*iverilog */
    initial
    begin            
        $dumpfile("wave.vcd");        //生成的vcd文件名称
        $dumpvars(0, testBench);    //tb模块名称
    end
/*iverilog */

endmodule
