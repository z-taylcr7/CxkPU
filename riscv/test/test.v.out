#! /c/Source/iverilog-install/bin/vvp
:ivl_version "12.0 (devel)" "(s20150603-1110-g18392a46)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision + 0;
:vpi_module "D:\iverilog\lib\ivl\system.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_sys.vpi";
:vpi_module "D:\iverilog\lib\ivl\vhdl_textio.vpi";
:vpi_module "D:\iverilog\lib\ivl\v2005_math.vpi";
:vpi_module "D:\iverilog\lib\ivl\va_math.vpi";
S_000001f09af85ec0 .scope module, "testBench" "testBench" 2 14;
 .timescale 0 0;
v000001f09afcf8e0_0 .var "A", 0 1;
v000001f09afcf980_0 .var "B", 0 1;
v000001f09afcfa20_0 .var "Cin", 0 0;
v000001f09afcfac0_0 .net "Cout", 0 0, v000001f09af52b20_0;  1 drivers
v000001f09afcfb60_0 .net "S", 0 1, v000001f09af861e0_0;  1 drivers
S_000001f09af86050 .scope module, "U" "adder" 2 20, 2 1 0, S_000001f09af85ec0;
 .timescale 0 0;
    .port_info 0 /INPUT 2 "X";
    .port_info 1 /INPUT 2 "Y";
    .port_info 2 /INPUT 1 "Cin";
    .port_info 3 /OUTPUT 1 "Cout";
    .port_info 4 /OUTPUT 2 "S";
v000001f09af76590_0 .net "Cin", 0 0, v000001f09afcfa20_0;  1 drivers
v000001f09af52b20_0 .var "Cout", 0 0;
v000001f09af861e0_0 .var "S", 0 1;
v000001f09af86280_0 .net "X", 0 1, v000001f09afcf8e0_0;  1 drivers
v000001f09afcf840_0 .net "Y", 0 1, v000001f09afcf980_0;  1 drivers
E_000001f09af3a1e0 .event anyedge, v000001f09af76590_0, v000001f09afcf840_0, v000001f09af86280_0;
    .scope S_000001f09af86050;
T_0 ;
    %wait E_000001f09af3a1e0;
    %load/vec4 v000001f09af86280_0;
    %pad/u 3;
    %load/vec4 v000001f09afcf840_0;
    %pad/u 3;
    %add;
    %load/vec4 v000001f09af76590_0;
    %pad/u 3;
    %add;
    %split/vec4 2;
    %store/vec4 v000001f09af861e0_0, 0, 2;
    %store/vec4 v000001f09af52b20_0, 0, 1;
    %jmp T_0;
    .thread T_0, $push;
    .scope S_000001f09af85ec0;
T_1 ;
    %pushi/vec4 0, 0, 2;
    %store/vec4 v000001f09afcf8e0_0, 0, 2;
    %pushi/vec4 2, 0, 2;
    %store/vec4 v000001f09afcf980_0, 0, 2;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v000001f09afcfa20_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 1, 0, 2;
    %store/vec4 v000001f09afcf8e0_0, 0, 2;
    %delay 10, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v000001f09afcfa20_0, 0, 1;
    %delay 10, 0;
    %pushi/vec4 0, 0, 2;
    %store/vec4 v000001f09afcf980_0, 0, 2;
    %delay 20, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v000001f09afcf8e0_0, 0, 2;
    %delay 30, 0;
    %pushi/vec4 0, 0, 1;
    %store/vec4 v000001f09afcfa20_0, 0, 1;
    %delay 40, 0;
    %pushi/vec4 3, 0, 2;
    %store/vec4 v000001f09afcf980_0, 0, 2;
    %delay 50, 0;
    %pushi/vec4 1, 0, 1;
    %store/vec4 v000001f09afcfa20_0, 0, 1;
    %end;
    .thread T_1;
    .scope S_000001f09af85ec0;
T_2 ;
    %vpi_call 2 42 "$monitor", $time, " ", " ", "part:%b  %b", v000001f09afcfb60_0, v000001f09afcfac0_0 {0 0 0};
    %delay 70, 0;
    %vpi_call 2 44 "$finish" {0 0 0};
    %end;
    .thread T_2;
    .scope S_000001f09af85ec0;
T_3 ;
    %vpi_call 2 49 "$dumpfile", "wave.vcd" {0 0 0};
    %vpi_call 2 50 "$dumpvars", 32'sb00000000000000000000000000000000, S_000001f09af85ec0 {0 0 0};
    %end;
    .thread T_3;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "test.v";
